modules/templates/handler.rb

```diff
@@ -7,23 +7,24 @@ module Proxy::Templates
   class Handler < ::Proxy::HttpRequest::ForemanRequest
     include Proxy::Log
 
-    def get_template(kind, token, static = false)
-      opts = { :token => token,
-               :url   => Proxy::Templates::Plugin.settings.template_url
-      }
-      opts[:static] = static if static
+    def get_template(kind, params, headers)
+      opts = params.merge(:url => Proxy::Templates::Plugin.settings.template_url)
       request = request_factory.create_get("/unattended/#{kind}", opts)
+      headers.each do |k, v|
+        request.add_field(k, v)
+      end
```

Comment:
Ok good point. I have amended one additional change because it was not working properly. Will amend the above two points later today.